components/script/dom/node.rs

```diff
@@ -2541,13 +2546,17 @@ impl<'a> ChildrenMutation<'a> {
     }
 
     /// Get the child that follows the added or removed children.
+    /// Currently only used when this mutation might force us to
+    /// restyle later children (see HAS_SLOW_SELECTOR_LATER_SIBLINGS and
+    /// Element's implementation of VirtualMethods::children_changed).
     pub fn next_child(&self) -> Option<&Node> {
         match *self {
             ChildrenMutation::Append { .. } => None,
             ChildrenMutation::Insert { next, .. } => Some(next),
             ChildrenMutation::Prepend { next, .. } => Some(next),
             ChildrenMutation::Replace { next, .. } => next,
             ChildrenMutation::ReplaceAll { .. } => None,
+            ChildrenMutation::Text => None,
```

Comment:
Of course probably before this we don't, so it should not block landing, but afterwards probably we still don't do the right thing... In particular, I'd expect servo to fail http://searchfox.org/mozilla-central/source/layout/reftests/bugs/534804-1.html.

If it does, mind filing an issue for it?